// wft-geodb-js-client/dist/api/LocaleApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _BadRequestResponse = _interopRequireDefault(require("../model/BadRequestResponse"));
var _CurrenciesResponse = _interopRequireDefault(require("../model/CurrenciesResponse"));
var _DateTimeResponse = _interopRequireDefault(require("../model/DateTimeResponse"));
var _ForbiddenResponse = _interopRequireDefault(require("../model/ForbiddenResponse"));
var _LanguagesResponse = _interopRequireDefault(require("../model/LanguagesResponse"));
var _LocalesResponse = _interopRequireDefault(require("../model/LocalesResponse"));
var _NotFoundResponse = _interopRequireDefault(require("../model/NotFoundResponse"));
var _TimeResponse = _interopRequireDefault(require("../model/TimeResponse"));
var _TimeZoneResponse = _interopRequireDefault(require("../model/TimeZoneResponse"));
var _TimeZonesResponse = _interopRequireDefault(require("../model/TimeZonesResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * GeoDB Cities API
                                                                                                                                                                                                                                                                                                                                                                                               * The GeoDB API focuses on getting global places and regions. Easily obtain country, region, and place data for use in your apps!  <ul>   <li>Filter places by name prefix, country, location, time-zone, and even minimum population.</li>   <li>Sort places by name, country code, elevation, and population - or any combination of these.</li>    <li>Get all country regions.</li> <li>Get all places in a given country or region.</li>   <li>     Display results in multiple languages.</li> <li>RESTful API adheres to industry best-practices, including     HATEOAS-style links to facilitate paging results.   </li>    <li>Backed by cloud-based load-balanced infrastructure.</li>   <li>Data is periodically refreshed from GeoNames and WikiData.</li> </ul> <p>Notes: <ul>   <li>     Since the database is periodically updated, this may <strong>very rarely</strong> result in certain places     being marked deleted (e.g., duplicates removed). By default, endpoints returning place data will exclude     places marked deleted. However, in the unlikely event that this occurs while your app is paging through a set     of affected results - and you care about the paged results suddenly changing underneath - specify      <tt>includeDeleted=SINCE_YESTERDAY</tt> (or <tt>SINCE_LAST_WEEK</tt> if you're really paranoid!).   </li> </ul> <hr/> <h3>Useful Resources</h3> <ul>   <li>     SDKs     <ul>       <li>         <a href='https://www.npmjs.com/package/wft-geodb-angular-client'>Angular</a>,          <a href='https://github.com/wirefreethought/geodb-sample-angular-app'>Sample App</a>       </li>       <li><a href='https://github.com/wirefreethought/geodb-java-client'>Java</a></li>       <li><a href='https://www.npmjs.com/package/wft-geodb-js-client'>JavaScript</a></li>     </ul>     <li><a href='swagger.json'>Swagger Docs</a></li>     <li><a href='http://creativecommons.org/licenses/by/3.0/'>Usage License</a></li>   </li>       </ul> 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* Locale service.
* @module api/LocaleApi
* @version 2.0.3
*/
var LocaleApi = /*#__PURE__*/function () {
  /**
  * Constructs a new LocaleApi. 
  * @alias module:api/LocaleApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function LocaleApi(apiClient) {
    _classCallCheck(this, LocaleApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Find currencies
   * Find currencies, filtering by optional criteria. If no criteria are set, you will get back all known currencies.
   * @param {String} countryId Currencies for this country id
   * @param {Object} opts Optional parameters
   * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
   * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
   * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/CurrenciesResponse} and HTTP response
   */
  _createClass(LocaleApi, [{
    key: "getCurrenciesUsingGETWithHttpInfo",
    value: function getCurrenciesUsingGETWithHttpInfo(countryId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'countryId' is set
      if (countryId === undefined || countryId === null) {
        throw new Error("Missing the required parameter 'countryId' when calling getCurrenciesUsingGET");
      }
      var pathParams = {};
      var queryParams = {
        'countryId': countryId,
        'hateoasMode': opts['hateoasMode'],
        'limit': opts['limit'],
        'offset': opts['offset']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _CurrenciesResponse["default"];
      return this.apiClient.callApi('/locale/currencies', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Find currencies
     * Find currencies, filtering by optional criteria. If no criteria are set, you will get back all known currencies.
     * @param {String} countryId Currencies for this country id
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/CurrenciesResponse}
     */
  }, {
    key: "getCurrenciesUsingGET",
    value: function getCurrenciesUsingGET(countryId, opts) {
      return this.getCurrenciesUsingGETWithHttpInfo(countryId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get languages
     * Get all supported languages
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/LanguagesResponse} and HTTP response
     */
  }, {
    key: "getLanguagesUsingGETWithHttpInfo",
    value: function getLanguagesUsingGETWithHttpInfo(opts) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'hateoasMode': opts['hateoasMode'],
        'limit': opts['limit'],
        'offset': opts['offset']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _LanguagesResponse["default"];
      return this.apiClient.callApi('/locale/languages', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get languages
     * Get all supported languages
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/LanguagesResponse}
     */
  }, {
    key: "getLanguagesUsingGET",
    value: function getLanguagesUsingGET(opts) {
      return this.getLanguagesUsingGETWithHttpInfo(opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get locales
     * Get all known locales
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/LocalesResponse} and HTTP response
     */
  }, {
    key: "getLocalesUsingGETWithHttpInfo",
    value: function getLocalesUsingGETWithHttpInfo(opts) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'hateoasMode': opts['hateoasMode'],
        'limit': opts['limit'],
        'offset': opts['offset']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _LocalesResponse["default"];
      return this.apiClient.callApi('/locale/locales', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get locales
     * Get all known locales
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/LocalesResponse}
     */
  }, {
    key: "getLocalesUsingGET",
    value: function getLocalesUsingGET(opts) {
      return this.getLocalesUsingGETWithHttpInfo(opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get time-zone date-time
     * Get time-zone date-time
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/DateTimeResponse} and HTTP response
     */
  }, {
    key: "getTimeZoneDateTimeUsingGETWithHttpInfo",
    value: function getTimeZoneDateTimeUsingGETWithHttpInfo(zoneId) {
      var postBody = null;
      // verify the required parameter 'zoneId' is set
      if (zoneId === undefined || zoneId === null) {
        throw new Error("Missing the required parameter 'zoneId' when calling getTimeZoneDateTimeUsingGET");
      }
      var pathParams = {
        'zoneId': zoneId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _DateTimeResponse["default"];
      return this.apiClient.callApi('/locale/timezones/{zoneId}/dateTime', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get time-zone date-time
     * Get time-zone date-time
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/DateTimeResponse}
     */
  }, {
    key: "getTimeZoneDateTimeUsingGET",
    value: function getTimeZoneDateTimeUsingGET(zoneId) {
      return this.getTimeZoneDateTimeUsingGETWithHttpInfo(zoneId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get time-zone time
     * Get time-zone time
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TimeResponse} and HTTP response
     */
  }, {
    key: "getTimeZoneTimeUsingGETWithHttpInfo",
    value: function getTimeZoneTimeUsingGETWithHttpInfo(zoneId) {
      var postBody = null;
      // verify the required parameter 'zoneId' is set
      if (zoneId === undefined || zoneId === null) {
        throw new Error("Missing the required parameter 'zoneId' when calling getTimeZoneTimeUsingGET");
      }
      var pathParams = {
        'zoneId': zoneId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TimeResponse["default"];
      return this.apiClient.callApi('/locale/timezones/{zoneId}/time', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get time-zone time
     * Get time-zone time
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TimeResponse}
     */
  }, {
    key: "getTimeZoneTimeUsingGET",
    value: function getTimeZoneTimeUsingGET(zoneId) {
      return this.getTimeZoneTimeUsingGETWithHttpInfo(zoneId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get time-zone
     * Get time-zone
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TimeZoneResponse} and HTTP response
     */
  }, {
    key: "getTimeZoneUsingGETWithHttpInfo",
    value: function getTimeZoneUsingGETWithHttpInfo(zoneId) {
      var postBody = null;
      // verify the required parameter 'zoneId' is set
      if (zoneId === undefined || zoneId === null) {
        throw new Error("Missing the required parameter 'zoneId' when calling getTimeZoneUsingGET");
      }
      var pathParams = {
        'zoneId': zoneId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TimeZoneResponse["default"];
      return this.apiClient.callApi('/locale/timezones/{zoneId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get time-zone
     * Get time-zone
     * @param {String} zoneId A time-zone id
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TimeZoneResponse}
     */
  }, {
    key: "getTimeZoneUsingGET",
    value: function getTimeZoneUsingGET(zoneId) {
      return this.getTimeZoneUsingGETWithHttpInfo(zoneId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get time-zones
     * Get all known time-zones
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TimeZonesResponse} and HTTP response
     */
  }, {
    key: "getTimezonesUsingGETWithHttpInfo",
    value: function getTimezonesUsingGETWithHttpInfo(opts) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'hateoasMode': opts['hateoasMode'],
        'limit': opts['limit'],
        'offset': opts['offset']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['UserSecurity'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TimeZonesResponse["default"];
      return this.apiClient.callApi('/locale/timezones', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get time-zones
     * Get all known time-zones
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.hateoasMode Include HATEOAS-style links in results (default to true)
     * @param {Number} opts.limit The maximum number of results to retrieve (default to 10)
     * @param {Number} opts.offset The zero-ary offset index into the results (default to 0)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TimeZonesResponse}
     */
  }, {
    key: "getTimezonesUsingGET",
    value: function getTimezonesUsingGET(opts) {
      return this.getTimezonesUsingGETWithHttpInfo(opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return LocaleApi;
}();
exports["default"] = LocaleApi;
