// wft-geodb-js-client/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "BadRequestResponse", {
  enumerable: true,
  get: function get() {
    return _BadRequestResponse["default"];
  }
});
Object.defineProperty(exports, "BaseCollectionResponse", {
  enumerable: true,
  get: function get() {
    return _BaseCollectionResponse["default"];
  }
});
Object.defineProperty(exports, "BaseCollectionResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseCollectionResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseResponse", {
  enumerable: true,
  get: function get() {
    return _BaseResponse["default"];
  }
});
Object.defineProperty(exports, "CountriesResponse", {
  enumerable: true,
  get: function get() {
    return _CountriesResponse["default"];
  }
});
Object.defineProperty(exports, "CountriesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _CountriesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "CountryDetails", {
  enumerable: true,
  get: function get() {
    return _CountryDetails["default"];
  }
});
Object.defineProperty(exports, "CountryResponse", {
  enumerable: true,
  get: function get() {
    return _CountryResponse["default"];
  }
});
Object.defineProperty(exports, "CountryResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _CountryResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "CountrySummary", {
  enumerable: true,
  get: function get() {
    return _CountrySummary["default"];
  }
});
Object.defineProperty(exports, "CurrenciesResponse", {
  enumerable: true,
  get: function get() {
    return _CurrenciesResponse["default"];
  }
});
Object.defineProperty(exports, "CurrenciesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _CurrenciesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "Currency", {
  enumerable: true,
  get: function get() {
    return _Currency["default"];
  }
});
Object.defineProperty(exports, "DateTimeResponse", {
  enumerable: true,
  get: function get() {
    return _DateTimeResponse["default"];
  }
});
Object.defineProperty(exports, "DateTimeResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DateTimeResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DistanceResponse", {
  enumerable: true,
  get: function get() {
    return _DistanceResponse["default"];
  }
});
Object.defineProperty(exports, "DistanceResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DistanceResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "ErrorCode", {
  enumerable: true,
  get: function get() {
    return _ErrorCode["default"];
  }
});
Object.defineProperty(exports, "ForbiddenResponse", {
  enumerable: true,
  get: function get() {
    return _ForbiddenResponse["default"];
  }
});
Object.defineProperty(exports, "GeoApi", {
  enumerable: true,
  get: function get() {
    return _GeoApi["default"];
  }
});
Object.defineProperty(exports, "LanguageDescriptor", {
  enumerable: true,
  get: function get() {
    return _LanguageDescriptor["default"];
  }
});
Object.defineProperty(exports, "LanguagesResponse", {
  enumerable: true,
  get: function get() {
    return _LanguagesResponse["default"];
  }
});
Object.defineProperty(exports, "LanguagesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _LanguagesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link["default"];
  }
});
Object.defineProperty(exports, "Locale", {
  enumerable: true,
  get: function get() {
    return _Locale["default"];
  }
});
Object.defineProperty(exports, "LocaleApi", {
  enumerable: true,
  get: function get() {
    return _LocaleApi["default"];
  }
});
Object.defineProperty(exports, "LocalesResponse", {
  enumerable: true,
  get: function get() {
    return _LocalesResponse["default"];
  }
});
Object.defineProperty(exports, "LocalesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _LocalesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "Metadata", {
  enumerable: true,
  get: function get() {
    return _Metadata["default"];
  }
});
Object.defineProperty(exports, "NotFoundResponse", {
  enumerable: true,
  get: function get() {
    return _NotFoundResponse["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlaceDetails", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlaceDetails["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlaceResponse", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlaceResponse["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlaceResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlaceResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlaceSummary", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlaceSummary["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlaceType", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlaceType["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlacesResponse", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlacesResponse["default"];
  }
});
Object.defineProperty(exports, "PopulatedPlacesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _PopulatedPlacesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "RegionDetails", {
  enumerable: true,
  get: function get() {
    return _RegionDetails["default"];
  }
});
Object.defineProperty(exports, "RegionResponse", {
  enumerable: true,
  get: function get() {
    return _RegionResponse["default"];
  }
});
Object.defineProperty(exports, "RegionResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _RegionResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "RegionSummary", {
  enumerable: true,
  get: function get() {
    return _RegionSummary["default"];
  }
});
Object.defineProperty(exports, "RegionsResponse", {
  enumerable: true,
  get: function get() {
    return _RegionsResponse["default"];
  }
});
Object.defineProperty(exports, "RegionsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _RegionsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "TimeResponse", {
  enumerable: true,
  get: function get() {
    return _TimeResponse["default"];
  }
});
Object.defineProperty(exports, "TimeResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _TimeResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "TimeZoneDescriptor", {
  enumerable: true,
  get: function get() {
    return _TimeZoneDescriptor["default"];
  }
});
Object.defineProperty(exports, "TimeZoneResponse", {
  enumerable: true,
  get: function get() {
    return _TimeZoneResponse["default"];
  }
});
Object.defineProperty(exports, "TimeZoneResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _TimeZoneResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "TimeZonesResponse", {
  enumerable: true,
  get: function get() {
    return _TimeZonesResponse["default"];
  }
});
Object.defineProperty(exports, "TimeZonesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _TimeZonesResponseAllOf["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _BadRequestResponse = _interopRequireDefault(require("./model/BadRequestResponse"));
var _BaseCollectionResponse = _interopRequireDefault(require("./model/BaseCollectionResponse"));
var _BaseCollectionResponseAllOf = _interopRequireDefault(require("./model/BaseCollectionResponseAllOf"));
var _BaseResponse = _interopRequireDefault(require("./model/BaseResponse"));
var _CountriesResponse = _interopRequireDefault(require("./model/CountriesResponse"));
var _CountriesResponseAllOf = _interopRequireDefault(require("./model/CountriesResponseAllOf"));
var _CountryDetails = _interopRequireDefault(require("./model/CountryDetails"));
var _CountryResponse = _interopRequireDefault(require("./model/CountryResponse"));
var _CountryResponseAllOf = _interopRequireDefault(require("./model/CountryResponseAllOf"));
var _CountrySummary = _interopRequireDefault(require("./model/CountrySummary"));
var _CurrenciesResponse = _interopRequireDefault(require("./model/CurrenciesResponse"));
var _CurrenciesResponseAllOf = _interopRequireDefault(require("./model/CurrenciesResponseAllOf"));
var _Currency = _interopRequireDefault(require("./model/Currency"));
var _DateTimeResponse = _interopRequireDefault(require("./model/DateTimeResponse"));
var _DateTimeResponseAllOf = _interopRequireDefault(require("./model/DateTimeResponseAllOf"));
var _DistanceResponse = _interopRequireDefault(require("./model/DistanceResponse"));
var _DistanceResponseAllOf = _interopRequireDefault(require("./model/DistanceResponseAllOf"));
var _Error = _interopRequireDefault(require("./model/Error"));
var _ErrorCode = _interopRequireDefault(require("./model/ErrorCode"));
var _ForbiddenResponse = _interopRequireDefault(require("./model/ForbiddenResponse"));
var _LanguageDescriptor = _interopRequireDefault(require("./model/LanguageDescriptor"));
var _LanguagesResponse = _interopRequireDefault(require("./model/LanguagesResponse"));
var _LanguagesResponseAllOf = _interopRequireDefault(require("./model/LanguagesResponseAllOf"));
var _Link = _interopRequireDefault(require("./model/Link"));
var _Locale = _interopRequireDefault(require("./model/Locale"));
var _LocalesResponse = _interopRequireDefault(require("./model/LocalesResponse"));
var _LocalesResponseAllOf = _interopRequireDefault(require("./model/LocalesResponseAllOf"));
var _Metadata = _interopRequireDefault(require("./model/Metadata"));
var _NotFoundResponse = _interopRequireDefault(require("./model/NotFoundResponse"));
var _PopulatedPlaceDetails = _interopRequireDefault(require("./model/PopulatedPlaceDetails"));
var _PopulatedPlaceResponse = _interopRequireDefault(require("./model/PopulatedPlaceResponse"));
var _PopulatedPlaceResponseAllOf = _interopRequireDefault(require("./model/PopulatedPlaceResponseAllOf"));
var _PopulatedPlaceSummary = _interopRequireDefault(require("./model/PopulatedPlaceSummary"));
var _PopulatedPlaceType = _interopRequireDefault(require("./model/PopulatedPlaceType"));
var _PopulatedPlacesResponse = _interopRequireDefault(require("./model/PopulatedPlacesResponse"));
var _PopulatedPlacesResponseAllOf = _interopRequireDefault(require("./model/PopulatedPlacesResponseAllOf"));
var _RegionDetails = _interopRequireDefault(require("./model/RegionDetails"));
var _RegionResponse = _interopRequireDefault(require("./model/RegionResponse"));
var _RegionResponseAllOf = _interopRequireDefault(require("./model/RegionResponseAllOf"));
var _RegionSummary = _interopRequireDefault(require("./model/RegionSummary"));
var _RegionsResponse = _interopRequireDefault(require("./model/RegionsResponse"));
var _RegionsResponseAllOf = _interopRequireDefault(require("./model/RegionsResponseAllOf"));
var _TimeResponse = _interopRequireDefault(require("./model/TimeResponse"));
var _TimeResponseAllOf = _interopRequireDefault(require("./model/TimeResponseAllOf"));
var _TimeZoneDescriptor = _interopRequireDefault(require("./model/TimeZoneDescriptor"));
var _TimeZoneResponse = _interopRequireDefault(require("./model/TimeZoneResponse"));
var _TimeZoneResponseAllOf = _interopRequireDefault(require("./model/TimeZoneResponseAllOf"));
var _TimeZonesResponse = _interopRequireDefault(require("./model/TimeZonesResponse"));
var _TimeZonesResponseAllOf = _interopRequireDefault(require("./model/TimeZonesResponseAllOf"));
var _GeoApi = _interopRequireDefault(require("./api/GeoApi"));
var _LocaleApi = _interopRequireDefault(require("./api/LocaleApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
